PR: https://github.com/odoo/odoo/pull/242504

From: ba061a17f65e62f8c64e727fba30f4690b2c47e1
From: alan-odoo
Date: 2026-02-13 14:01:29

Breaking data model changes scores: del:3 + add:1, change matches:
-    _inherit = ["mail.thread"]
-    activity_ids = fields.One2many('mail.activity', 'calendar_event_id', string='Activities')
+    meeting_activity_ids = fields.One2many('mail.activity', 'calendar_event_id', string='Meeting Activities')

Total Changes: 21

[IMP] calendar, *: allows calendar events to have activities

* = crm, hr_holidays

This commit allows to link activities to calendar.event. From now on, calendar
events can be linked to all types of activities from an activity view and the
advanced calendar event form.

The calendar.event model had already an activities menu allowing to display
the today's meetings. This has been adapted to display the default activities
menu generated by calendar.event's inheritance of mail.activity.mixin.

The activity_id field of calendar.event has been renamed
meeting_activity_ids. This fields allows models to have meeting activities. As
it had the same name as the one inherited from mail.activity.mixing model,
making not possible to profit of this last one, the field had to be renamed.

Enterprise PR: https://github.com/odoo/enterprise/pull/103469
Upgrade PR: https://github.com/odoo/upgrade/pull/9344

Task-5429852

Part-of: odoo/odoo#242504
Related: odoo/enterprise#103469
Related: odoo/upgrade#9344
Signed-off-by: Jérémy Hennecart (jeh) <jeh@odoo.com>

================================= pseudo patch: =================================

--- a/addons/calendar/models/calendar_event.py
+++ b/addons/calendar/models/calendar_event.py
@@ -72,7 +72,7 @@ class CalendarEvent(models.Model):
     _name = 'calendar.event'
     _description = "Calendar Event"
     _order = "start desc"
-    _inherit = ["mail.thread"]
+    _inherit = ["mail.thread", "mail.activity.mixin"]
     _mail_post_access = 'read'
     _systray_view = 'calendar'
 
@@ -199,7 +199,7 @@ class CalendarEvent(models.Model):
         'Document Model Name', related='res_model_id.model', readonly=True, store=True)
     res_model_name = fields.Char(related='res_model_id.name')
     # messaging
-    activity_ids = fields.One2many('mail.activity', 'calendar_event_id', string='Activities')
+    meeting_activity_ids = fields.One2many('mail.activity', 'calendar_event_id', string='Meeting Activities')
     # attendees
     attendee_ids = fields.One2many(
         'calendar.attendee', 'event_id', 'Participant')
@@ -600,14 +600,14 @@ class CalendarEvent(models.Model):
         # Prevent sending update notification when _inverse_dates is called
         self = self.with_context(is_calendar_event_new=True)
         defaults = self.browse().default_get([
-            'activity_ids', 'allday', 'description', 'name', 'partner_ids',
+            'meeting_activity_ids', 'allday', 'description', 'name', 'partner_ids',
             'res_model_id', 'res_id', 'start', 'user_id',
         ])
 
         vals_list = [  # Else bug with quick_create when we are filter on an other user
             {
                 **vals,
-                'activity_ids': vals.get('activity_ids', defaults.get('activity_ids')),
+                'meeting_activity_ids': vals.get('meeting_activity_ids', defaults.get('meeting_activity_ids')),
                 'allday': vals.get('allday', defaults.get('allday')),
                 'description': vals.get('description', defaults.get('description')),
                 'name': vals.get('name', defaults.get('name')),
@@ -638,7 +638,7 @@ class CalendarEvent(models.Model):
         if meeting_activity_types:
             for values in vals_list:
                 # created from calendar: try to create an activity on the related record
-                if values['activity_ids'] and not existing_event:
+                if values['meeting_activity_ids'] and not existing_event:
                     continue
                 res_model = all_models.filtered(lambda m: m.id == values['res_model_id'])
                 res_id = values['res_id']
@@ -668,7 +668,7 @@ class CalendarEvent(models.Model):
                     activity_vals['date_deadline'] = self._get_activity_deadline_from_start(fields.Datetime.from_string(values['start']), values['allday'])
                 if values['user_id']:
                     activity_vals['user_id'] = values['user_id']
-                values['activity_ids'] = [(0, 0, activity_vals)]
+                values['meeting_activity_ids'] = [(0, 0, activity_vals)]
 
         self._set_videocall_location(vals_list)
 
@@ -733,7 +733,7 @@ class CalendarEvent(models.Model):
         # complete
         to_sync_activities = self.browse()
         for event, event_values in zip(events, vals_list):
-            if any(command[0] != 0 for command in event_values.get('activity_ids') or []):
+            if any(command[0] != 0 for command in event_values.get('meeting_activity_ids') or []):
                 to_sync_activities += event
         to_sync_activities._sync_activities(fields={f for vals in vals_list for f in vals})
 
@@ -1235,7 +1235,7 @@ class CalendarEvent(models.Model):
     def _sync_activities(self, fields):
         # update activities
         for event in self:
-            if event.activity_ids:
+            if event.meeting_activity_ids:
                 activity_values = {}
                 if 'name' in fields:
                     activity_values['summary'] = event.name
@@ -1247,7 +1247,7 @@ class CalendarEvent(models.Model):
                 if 'user_id' in fields:
                     activity_values['user_id'] = event.user_id.id
                 if activity_values.keys():
-                    event.activity_ids.write(activity_values)
+                    event.meeting_activity_ids.write(activity_values)
 
     @api.model
     def _get_activity_deadline_from_start(self, start, allday):

--- a/addons/calendar/models/mail_activity.py
+++ b/addons/calendar/models/mail_activity.py
@@ -41,7 +41,7 @@ class MailActivity(models.Model):
             'default_res_model': self.env.context.get('default_res_model'),
             'default_name': self.res_name,
             'default_description': self.note if not is_html_empty(self.note) else '',
-            'default_activity_ids': [(6, 0, self.ids)],
+            'default_meeting_activity_ids': [(6, 0, self.ids)],
             'default_partner_ids': self.user_id.partner_id.ids,
             'default_user_id': self.user_id.id,
             'initial_date': self.date_deadline,

--- a/addons/calendar/models/res_users.py
+++ b/addons/calendar/models/res_users.py
@@ -160,6 +160,7 @@ class ResUsers(models.Model):
                 'id': self.env['ir.model']._get('calendar.event').id,
                 'type': 'meeting',
                 'name': meeting_label,
+                'is_today_meetings': True,
                 'model': 'calendar.event',
                 'icon': modules.module.get_module_icon(EventModel._original_module),
                 'domain': [('active', 'in', [True, False])],
